platforms/nuttx/src/px4/common/usr_hrt.cpp

```diff
@@ -61,6 +61,7 @@
 #include <sys/boardctl.h>
 
 static px4_task_t g_usr_hrt_task = -1;
+static px4_sem_t *g_wait_sem;
```

Comment:
The user does not need to know that the synchronization is done via a semaphore, or anything about how it is done for that matter.

Better to define a handle for this, like typedef void* px4_hrt_handle_t or something similar, to hide the details.

boardctl(HRT_REGISTER, (uintptr_t)&g_hrt_client_handle); or something like that.